src/bun.js/bindings/NodeValidator.cpp

```diff
@@ -378,19 +391,32 @@ JSC_DEFINE_HOST_FUNCTION(jsFunction_validateEncoding, (JSC::JSGlobalObject * glo
     JSC::VM& vm = globalObject->vm();
     auto scope = DECLARE_THROW_SCOPE(vm);
 
-    auto data = callFrame->argument(0);
     auto encoding = callFrame->argument(1);
 
     auto normalized = WebCore::parseEnumeration<BufferEncodingType>(*globalObject, encoding);
-    // no check to throw ERR_UNKNOWN_ENCODING ? it's not in node but feels like it would be apt here
-
-    auto length = data.get(globalObject, Identifier::fromString(vm, "length"_s));
-    RETURN_IF_EXCEPTION(scope, {});
-    auto length_num = length.toNumber(globalObject);
-    RETURN_IF_EXCEPTION(scope, {});
-    if (normalized == BufferEncodingType::hex && std::fmod(length_num, 2.0) != 0) {
-        return Bun::ERR::INVALID_ARG_VALUE(scope, globalObject, "encoding"_s, encoding, makeString("is invalid for data of length "_s, length_num));
+    if (normalized == BufferEncodingType::hex) {
+        auto data = callFrame->argument(0);
+
+        size_t length = 0;
+        if (data.isString()) {
+            length = data.toString(globalObject)->length();
+        } else if (auto* view = JSC::jsDynamicCast<JSC::JSArrayBufferView*>(data)) {
+            length = view->length();
+        } else if (auto* buffer = JSC::jsDynamicCast<JSC::JSArrayBuffer*>(data)) {
+            if (auto* impl = buffer->impl()) {
```

Comment:
when would the first condition be true but this one be false?

Reply:
`ArrayBuffer` != `ArrayBufferView`, `Uint8Array` would fit `ArrayBufferView` but not `ArrayBuffer`.

Reply:
yes, i meant specifically `if (auto* buffer =` and `if (auto* impl =`